### PR TITLE
[Fixes #56] Remove PACs from this repository

### LINKED ISSUE #56: (Re)move support for PAC from this repository
The objective for quite a long time is to remove PAC maintenance burden from this repo toward [atsamx7x-pac](https://github.com/atsams-rs/atsamx7x-pac) that would in future fully automatically release following releases of `svd2rust`.

So far, the code on that repo regenerates PACs that when used do `svd2rust` heavy lifting on the user side, so we are able to keep very lightweight repository with copy of SVD files for reference.

### PR DESCRIPTION
**Pull request by michalfita**
# Purpose
This PR removes PACs (you can skip reviewing first commit) from this repository in favour of [`atsamx7x-pac`](/atsams-rs/atsamx7x-pac) repository. The real revolutions happens there as that repo doesn't keep generated PACs (as they're huge) but they're generated via `build.rs` for the user on-the-fly during build.

Local use of `hal` requires [`atsamx7x-pac`](/atsams-rs/atsamx7x-pac) to be cloned aside. Is there better method? I've seen someone used `[patch.crates-io]` instead of local patch in dependencies, but I haven't tried yet how this works the directory pointed at doesn't exists, hence draft.

If any of you would have time to play with this branch (or alpha PACs from `creates.io`) it would be great.

# Scope
* Switch `Cargo.toml`s to the new model and use pre-release of PACs.
* Fixes examples so they build & **link** again.

# To do
* _Either_\
  fix the CI but checking out [`atsamx7x-pac`](/atsams-rs/atsamx7x-pac) aside \
  _or_\
  find way for [`atsamx7x-pac`](/atsams-rs/atsamx7x-pac) clone to be optional.

### CODE REVIEW

**Comment by martinmortsell** [comment]
`tools/update-pacs.sh` should be removed, as we're no longer updating pacs in this repo.

What's the point of the relative paths? Isn't it easier to just use the crates.io versions of the pacs?

**Comment by michalfita** [comment]
> What's the point of the relative paths? Isn't it easier to just use the crates.io versions of the pacs?

Checked out repo still should let someone work on PACs... I think `[patch.crates-io]` does the job, but didn't have enough time yesterday to finish testing this.

Plus [atsamx7x-pac](https://github.com/atsams-rs/atsamx7x-pac) would need CI that actually builds HAL against them for testing.

**Comment by michalfita** [comment]
CI is broken by some HAL issues that newer compiler reveals now as warnings.
Raised #62.

**Comment by martinmortsell** [comment]
I see what is happening.
The CI is broken for this PR specifically as the `workflow.yml` file is outdated and `rust-toolchain.toml` does not contain a version specification.

These warnings have been added after the release of rustc 1.63, we haven't tested that before, so we haven't seen the warnings before. I did see them when working on #60  .

TLDR: rebase on top of development and CI should work again.

**Comment by michalfita** [comment]
In the light of lack of resistance for quite long period of time I merge this.
